libs/agno/agno/agent/agent.py

```diff
@@ -2657,6 +2753,32 @@ def get_system_message(self, session_id: str, user_id: Optional[str] = None) ->
         if self.name is not None and self.add_name_to_instructions:
             additional_information.append(f"Your name is: {self.name}.")
 
+        # 3.2.4 Add information about agentic filters if enabled
+        if self.knowledge is not None and self.enable_agentic_filters:
+            valid_filters = getattr(self.knowledge, "valid_metadata_filters", None)
+            if valid_filters:
+                valid_filters_str = ", ".join(valid_filters)
+                additional_information.append(
+                    dedent(f"""
+                    The knowledge base contains documents with these metadata filters: {valid_filters_str}.
+                    When searching the knowledge base, you MUST use these filters to narrow down results.
+                    Always use filters when the user query indicates specific metadata.
```

Comment:
```suggestion
                    Always use filters when the user query indicates specific metadata.
```
We have filters required on the function, so it is fine